editor/profile_picture.cpp

```diff
@@ -0,0 +1,54 @@
+#include "editor/profile_picture.hpp"
+
+#include "platform/remote_file.hpp"
+#include "platform/platform.hpp"
+#include "platform/settings.hpp"
+
+#include "filesystem"
+
+// Example pic_url: https://www.openstreetmap.org/rails/active_storage/representations/redirect/eyJfcmFpbHMiOnsiZGF0YSI6MzM1MjcxMDUsInB1ciI6ImJsb2JfaWQifX0=--957a2a99111c97d4b9f4181003c3e69713a221c6/eyJfcmFpbHMiOnsiZGF0YSI6eyJmb3JtYXQiOiJwbmciLCJyZXNpemVfdG9fbGltaXQiOlsxMDAsMTAwXX0sInB1ciI6InZhcmlhdGlvbiJ9fQ==--473f0190a9e741bb16565db85fe650d7b0a9ee69/cat.png
+// Example hash: 473f0190a9e741bb16565db85fe650d7b0a9ee69
+namespace editor
+{
+  std::string ProfilePicture::download(const std::string& pic_url)
```

Comment:
Is there any generic "download file" api already somewhere? If no, it would be better to add a generic "download a file" API, with an optional hint of expected MIME file type if necessary, so it can be reused to download anything.